### PR TITLE
Migrate Site.test to Typescript

### PR DESCRIPTION
**Pull request by yiwen101**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [x] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
Mostly some type cleaning and method arguments alignment. Working on #1913.

**Anything you'd like to highlight/discuss:**

**Testing instructions:**

**Proposed commit message: (wrap lines at 72 characters)**
Migrate Site.test to Typescript

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [ ] Linked all related issues
- [ ] No unrelated changes 

---

**Reviewer checklist:**

Indicate the [SEMVER](https://semver.org/) impact of the PR:
- [ ] Major (when you make incompatible API changes)
- [ ] Minor (when you add functionality in a backward compatible manner)
- [x] Patch (when you make backward compatible bug fixes)

At the end of the review, please label the PR with the appropriate label: `r.Major`, `r.Minor`, `r.Patch`.

Breaking change release note preparation (if applicable):
- To be included in the release note for any feature that is made obsolete/breaking

> Give a brief explanation note about:
>   - what was the old feature that was made obsolete
>   - any replacement feature (if any), and
>   - how the author should modify his website to migrate from the old feature to the replacement feature (if possible).

### CODE REVIEW

File: packages/core/test/unit/Site.test.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
+
+import {
+  INDEX_MD_DEFAULT, PAGE_NJK, SITE_JSON_DEFAULT, getDefaultTemplateFileFullPath,
+} from './utils/data';
+import { SiteConfig } from '../../src/Site/SiteConfig';
 
 const DEFAULT_TEMPLATE = 'default';
+const mockFs = fs as any;
+const mockGhPages = ghpages as any;
+type SiteArguements = [string, string, string, undefined, undefined, any, boolean, () => void];
```
Comment (LamJiuFong) [inline packages/core/test/unit/Site.test.ts:15]:
```suggestion
type SiteArguments = [string, string, string, undefined, undefined, any, boolean, () => void];
```

**Review by LamJiuFong — COMMENTED** [review]
LGTM!

File: packages/core/test/unit/Site.test.ts [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+const siteArguements: SiteArguements = ['./', '_site', '', undefined, undefined, undefined, false, () => {}];
```
[the remark this answers is not in the document]
Comment (yiwen101) [reply packages/core/test/unit/Site.test.ts]: Thanks for spotting this. I have correct the name in a amendment commit.

**Comment by kaixin-hc** [comment]
@yiwen101 Could you also remove the .js file in this PR?

**Comment by yiwen101** [comment]
> @yiwen101 Could you also remove the .js file in this PR? When pulling your branch I still see it

It is removed(renamed).
I think you still see it because I modify the gitnore to ignore it?

**Review by kaixin-hc — APPROVED** [review]
okay! Sorry about earlier - my editor i think cached the file so it looked like the .js file was being tracked and not properly ignored. But it is ignored, so all looks good to me : )

**Pull request merged**